src/lib/sw_engine/tvgSwRasterNeon.h

```diff
@@ -91,36 +91,32 @@ static inline bool neonRasterTranslucentRle(SwSurface* surface, const SwRleData*
 
 static inline bool neonRasterTranslucentRect(SwSurface* surface, const SwBBox& region, uint32_t color)
 {
-    uint32_t *buffer = surface->buffer + (region.min.y * surface->stride) + region.min.x;
-    uint32_t h = (uint32_t)(region.max.y - region.min.y);
-    uint32_t w = (uint32_t)(region.max.x - region.min.x);
-
-    uint32_t ialpha = 255 - surface->blender.alpha(color);
+    auto buffer = surface->buffer + (region.min.y * surface->stride) + region.min.x;
+    auto h = static_cast<uint32_t>(region.max.y - region.min.y);
+    auto w = static_cast<uint32_t>(region.max.x - region.min.x);
+    auto ialpha = 255 - surface->blender.alpha(color);
 
-    uint8x8_t vColor = (uint8x8_t) vdup_n_u32(color);
-    uint8x8_t vIalpha = (uint8x8_t) vdup_n_u8((uint8_t) ialpha);
-
-    uint8x8_t *vDst = NULL;
+    auto vColor = static_cast<uint8x8_t>(vdup_n_u32(color));
+    auto vIalpha = static_cast<uint8x8_t>(vdup_n_u8((uint8_t) ialpha));
+    uint8x8_t* vDst = nullptr;
 
     for (uint32_t y = 0; y < h; ++y) {
-        uint32_t align = 0;
-        uint32_t *dst = &buffer[y * surface->stride];
+
+        auto *dst = &buffer[y * surface->stride];
 
         if ((((uint32_t) dst) & 0x7) != 0) {
-            vDst = (uint8x8_t*)(dst+1);
-            align = 1;
+            //fill not aligned byte
+            *dst = color + ALPHA_BLEND(*dst, ialpha);
+            vDst = (uint8x8_t*) (dst + 1);
         } else {
             vDst = (uint8x8_t*) dst;
         }
 
-        uint32_t iterations = w / 2;
-        uint32_t left = w % 2;
-
-        if (align) *dst = color + ALPHA_BLEND(*dst, ialpha);
-
-        for (uint32_t x = 0; x < iterations; ++x) vDst[x] = vadd_u8(vColor, ALPHA_BLEND_NEON(vDst[x], vIalpha));
+        for (uint32_t x = 0; x <  w / 2; ++x)
```

Comment:
there is a problem in the case where `w` is even and the `dst` address is not aligned.
lets take `w=4`:
- first we fill the not aligned bytes (line 109) - 1 pixel
- in the for loop (line 115) `x` can be 0 and 1 so we try to fill another 4 pixels, but there were only 4 in total

Reply:
I think that is another story out of this patch boundary. Please make a patch.

Reply:
I opened #747.
If you first merge your, I'll solve conflicts in mine